packages/common/src/project/load.ts

```diff
@@ -3,42 +3,60 @@
 
 import fs from 'fs';
 import path from 'path';
+import {NodeVM, VMScript} from '@subql/x-vm2';
 import {plainToClass} from 'class-transformer';
 import {validateSync} from 'class-validator';
 import yaml from 'js-yaml';
 import {ChainTypes} from './models';
 import {ProjectManifestVersioned, VersionedProjectManifest} from './versioned';
 
-export function loadFromJsonOrYaml(file: string): unknown {
-  const {ext} = path.parse(file);
+export function loadFromFile(filePath: string): unknown {
+  const {base, ext} = path.parse(filePath);
 
-  if (ext !== '.yaml' && ext !== '.yml' && ext !== '.json') {
+  if (ext !== '.yaml' && ext !== '.yml' && ext !== '.json' && ext !== '.js') {
     throw new Error(`Extension ${ext} not supported`);
   }
 
-  const rawContent = fs.readFileSync(file, 'utf-8');
-  return yaml.load(rawContent);
+  if (ext === '.js') {
+    const vm = new NodeVM({
+      console: 'inherit',
+      wasm: false,
+      sandbox: {},
+      require: {
+        external: true,
+        context: 'sandbox',
+      },
+      wrapper: 'commonjs',
+      sourceExtensions: ['js', 'cjs'],
+    });
+
+    const script = new VMScript(`module.exports = require('./${base}');`, filePath);
+
+    return vm.run(script) as unknown;
```

Comment:
you can add some logic here to check the keys are in our whitelist.

Reply:
This logic exists in a parseChainTypes() and I have written a test to check that an exception is thrown if extra keys are added.